src/cursor_on_target.py

```diff
@@ -0,0 +1,183 @@
+#!/usr/bin/env python3
+"""
+cursor_on_target.py
+
+This module implements Cursor on Target (CoT) functionality for the OpenAthena project.
+It provides functions to create, format, and send CoT messages over a UDP multicast network.
+
+CoT is a XML-based schema for sharing location information between different systems.
+This implementation focuses on sending location data from drone imagery analysis.
+
+The module uses the following libraries:
+- xml.etree.ElementTree for XML creation
+- uuid for generating unique identifiers
+- time for timestamping
+- socket for network communication
+
+Usage:
+    Import this module and use the create_and_send_cot function to send CoT messages
+    with resolved target information.
+
+Note:
+    This implementation uses UDP multicast by default, which may not be suitable
+    for all network environments. Ensure your network supports multicast before use.
+"""
+
+import xml.etree.ElementTree as ET
+import uuid
+import time
+import socket
+#import struct
+import socket
+import logging
+
+"""
+    Generate a unique identifier for CoT messages.
+
+    Returns:
+    str: A unique identifier string prefixed with 'OpenAthena-'.
+
+    Note:
+    This function uses UUID4 to ensure uniqueness across multiple instances
+    and executions of the software.
+"""
+def create_uid():
+    return f"OpenAthena-{uuid.uuid4()}"
```

Comment:
This would technically work, but the marker on ATAK will contain mostly random letters and numbers which are difficult for humans to disambiguate verbally. 

A convention we've used so far to solve this is:
OpenAthena-DeviceUID-CalculationSerialNumber

Where the serial number starts at 1 and increases monotonically with a value that persists even across app restarts.

In the [latest update of OA Android](https://github.com/Theta-Limited/OpenAthenaAndroid/issues/141), iOS, and Core, we've [switched to a new convention for the device UID](https://github.com/Theta-Limited/OpenAthenaAndroid/commit/67955f9cb9ec40070e07787ad4fafdfeca21e880) which uses a hash of the device MAC address. This value indexes randomly into a [NATO-phonetic alphabet letter code word](https://en.wikipedia.org/wiki/NATO_phonetic_alphabet) followed by two digits for 2600 total possible permutations. Very unlikely to experience a collision in any real-world setting, yet easy for humans to identify and disambiguate verbally (even with minimal English language skills)

E.g: `OpenAthena-Alfa01-1`

![Screenshot_20240620-185422](https://github.com/Theta-Limited/OpenAthenaAndroid/assets/25494111/7e457256-d565-42b5-a813-7d92366885a1)